tests/test_cli_client.cpp

```diff
@@ -3017,11 +3200,104 @@ TEST_F(ClientAlias, fails_when_name_clashes_with_command_alias)
 
 TEST_F(ClientAlias, fails_when_name_clashes_with_command_name)
 {
-    EXPECT_CALL(mock_daemon, info(_, _, _)).Times(AtMost(1)).WillRepeatedly(info_function);
+    EXPECT_CALL(mock_daemon, info(_, _, _)).Times(AtMost(1)).WillRepeatedly(make_info_function());
 
     std::stringstream cerr_stream;
     send_command({"alias", "primary:command", "list"}, trash_stream, cerr_stream);
 
     ASSERT_THAT(cerr_stream.str(), Eq("Alias name 'list' clashes with a command name\n"));
 }
+
+TEST_F(ClientAlias, execAliasRewritesMountedDir)
+{
+    std::string alias_name{"an_alias"};
+    std::string instance_name{"primary"};
+    std::string cmd{"pwd"};
+
+    QDir current_dir{QDir::current()};
+    std::string source_dir{(current_dir.canonicalPath()).toStdString()};
+    std::string target_dir{"/home/ubuntu/dir"};
+
+    EXPECT_CALL(mock_daemon, info(_, _, _)).Times(AtMost(1)).WillRepeatedly(make_info_function(source_dir, target_dir));
+
+    populate_db_file(AliasesVector{{alias_name, {instance_name, cmd, true}}});
+
+    REPLACE(ssh_channel_request_exec, ([&target_dir, &cmd](ssh_channel, const char* raw_cmd) {
+                EXPECT_THAT(raw_cmd, StartsWith("'cd' '" + target_dir + "/'"));
+                EXPECT_THAT(raw_cmd, HasSubstr("&&"));
+                EXPECT_THAT(raw_cmd, EndsWith("'" + cmd + "'"));
+
+                return SSH_OK;
+            }));
+
+    mp::SSHInfoReply ssh_info_response = make_fake_ssh_info_response(instance_name);
+
+    EXPECT_CALL(mock_daemon, ssh_info(_, _, _))
+        .WillOnce([&ssh_info_response](grpc::ServerContext* context, const mp::SSHInfoRequest* request,
+                                       grpc::ServerWriter<multipass::SSHInfoReply>* server) {
+            server->Write(ssh_info_response);
+
+            return grpc::Status{};
+        });
+
+    EXPECT_EQ(send_command({alias_name}), mp::ReturnCode::Ok);
+}
+
+struct NotDirRewriteTestsuite : public ClientAlias, public WithParamInterface<std::pair<bool, QString>>
+{
+};
+
+TEST_P(NotDirRewriteTestsuite, execAliasDoesNotRewriteMountedDir)
+{
+    auto [use_no_map_argument, source_qdir] = GetParam();
+
+    std::string alias_name{"an_alias"};
+    std::string instance_name{"primary"};
+    std::string cmd{"pwd"};
+
+    QDir current_dir{QDir::current()};
+    std::string source_dir{source_qdir.toStdString()};
+    std::string target_dir{"/home/ubuntu/dir"};
+
+    EXPECT_CALL(mock_daemon, info(_, _, _)).Times(AtMost(1)).WillRepeatedly(make_info_function(source_dir, target_dir));
```

Comment:
Indeed, being called 0 or 1 time depends on the parameters of the test: it's not called when the `--no-map-working-directory` is present, and called otherwise.